File: Python.PassDetection/src/services/feature_engineer.py
from typing import List

from src.domain.inferences import Feature, InputData


class FeatureEngineer:
    def __init__(self):
        self.features: List[Feature] = []

    def add_feature(self, feature_calculator: Feature):
        self.features.append(feature_calculator)

    def engineer(self, input_data: InputData) -> List[float]:
        inputs = []
        for feature in self.features:
            inputs.extend(feature.calculate(input_data))

        return inputs

    @property
    def feature_size(self) -> int:
        return len(self.features)

File: VUInteractions/Assets/_Project/MachineLearning/Features/FeatureEngineer.cs
using System.Collections.Generic;

namespace Tactive.MachineLearning.Features
{
	public class FeatureEngineer<T>
	{
		readonly List<Feature<T>> _features = new();

		public int FeatureSize => _features.Count;

		public void AddFeature(Feature<T> featureCalculator)
		{
			_features.Add(featureCalculator);
		}

		public IReadOnlyList<float> Engineer(T inputData)
		{
			var inputs = new List<float>();

			foreach (var feature in _features)
				inputs.AddRange(feature.Calculate(inputData));

			return inputs;
		}
	}
}